### PR TITLE
Added kind and type columns for all trigger events and created view trigger_events

### PR DESCRIPTION
**Pull request by johnnieskywalker**

### CODE REVIEW

File: src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql [hunk 1]
```diff
@@ -0,0 +1,49 @@
+ALTER TABLE automation_bot.trigger_added_events
+	ADD column kind character varying (66);
+		
+UPDATE automation_bot.trigger_added_events 
+	SET kind = 'stop-loss'
```
Comment (adamskrodzki) [inline src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql:5]: what is kind ? doesn't this table already have triggetTypeId?

File: src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		
+UPDATE automation_bot.trigger_executed_events  
+	SET kind = 'stop-loss'
+	WHERE kind = null ;
+
+ALTER TABLE automation_bot.trigger_added_events 
+	ADD column event_type character varying (66);
+
+UPDATE automation_bot.trigger_added_events 
+	SET event_type = 'added'
```
Comment (adamskrodzki) [inline src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql:26]: this do not need to sit in table, you can do this on view level

File: src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+UPDATE automation_bot.trigger_executed_events
+	SET event_type = 'executed'
+	WHERE kind = null ;	
+
+CREATE view api.trigger_events as (
+	SELECT id , trigger_id , cdp_id, tx_id , block_id, kind , event_type  from automation_bot.trigger_added_events tae
+	UNION
+	SELECT  id , trigger_id , cdp_id, tx_id , block_id, kind , event_type  from automation_bot.trigger_removed_events tre 
+	UNION 
+	SELECT  id , trigger_id , cdp_id, tx_id , block_id, kind , event_type  from automation_bot.trigger_executed_events tee
```
Comment (adamskrodzki) [inline src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql:48]: `SELECT id , trigger_id , cdp_id, tx_id , block_id, kind , 'added' as event_type  from automation_bot.trigger_added_events tae
	UNION
	SELECT  id , trigger_id , cdp_id, tx_id , block_id, kind , 'removes' as event_type  from automation_bot.trigger_removed_events tre 
	UNION 
	SELECT  id , trigger_id , cdp_id, tx_id , block_id, kind , 'executed' as event_type  from automation_bot.trigger_executed_events tee`

File: src/borrow/transformers/automationBotTransformer.ts [hunk 4]
```diff
@@ -31,9 +31,9 @@ async function handleTriggerAdded(
 
   await services.tx.none(
     `INSERT INTO automation_bot.trigger_added_events(
-      trigger_id, cdp_id, command_address, trigger_data, log_index, tx_id, block_id
+      trigger_id, cdp_id, command_address, trigger_data, log_index, tx_id, block_id, kind, event_type
```
Comment (adamskrodzki) [inline src/borrow/transformers/automationBotTransformer.ts:34]: what execly kind is needed for ?
triggerData includes triggerType and also command_address fully determines triggerType

**Review by adamskrodzki — APPROVED** [review]

File: src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql (hunk 3 above)
Comment (piekczyk) [reply src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql:48]: If we want to be consistent with other event interfaces you should include here urn, timestamp, log_index, hash - there may be more to include beside those. Take a look how query for multiply history view looks like.

https://github.com/OasisDEX/oasis-borrow/blob/dev/features/vaultHistory/vaultMultiplyHistory.ts#L14

With current column set (plus those that I mentioned) we will be able to show on the UI that event happen but without any details.

File: src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql (hunk 2 above)
Comment (piekczyk) [reply src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql:26]: If we use @Adam suggestion lets do it in the way that will be consistent with other history views. It means that kind should have format more or less like this STOP_LOSS_ADDED, STOP_LOSS_REMOVED, STOP_LOSS_EXECUTED.

File: src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql (hunk 3 above)
Comment (piekczyk) [reply src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql:48]: UNION does not ensure proper handling of ids which should be unique here. I guess we don't need to select ids here, rather generating unique ones in the scope of this view.

image

File: src/borrow/transformers/automationBotTransformer.ts [hunk 5]
```diff
@@ -86,9 +86,9 @@ async function handleTriggerExecuted(
   };
   await services.tx.none(
     `INSERT INTO automation_bot.trigger_executed_events(
-        trigger_id, cdp_id, vault_closed_event, log_index, tx_id, block_id
+        trigger_id, cdp_id, vault_closed_event, log_index, tx_id, block_id, kind, event_type
     ) VALUES (
-        \${trigger_id}, \${cdp_id}, \${close_event_id}, \${log_index}, \${tx_id}, \${block_id}
+        \${trigger_id}, \${cdp_id}, \${close_event_id}, \${log_index}, \${tx_id}, \${block_id}, 'stop-loss', 'executed'
```
Comment (johnnieskywalker) [inline src/borrow/transformers/automationBotTransformer.ts:91]: this inserts will no longer work, will fix

File: src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql (hunk 3 above)
Comment (johnnieskywalker) [reply src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql]: urn is fetched from blockchain see example here
https://github.com/OasisDEX/oasis-borrow-cache/blob/dev/src/borrow/dependencies/getUrnForCdp.ts

so I could add urn for each for ex. `trigger_added` record when the transformer is creating those I'd have to add urn column in that table I believe unless there's SQL way to get urn for cdpId

File: src/borrow/migrations/022-automation-events-ddl-kind-type.sql [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+CREATE VIEW api.trigger_events AS (
+	SELECT tae.id , trigger_id , cdp_id, tae.log_index, tx.hash , b."number", b."timestamp" , 'added' AS event_type, tae.command_address,  alias.kind  from automation_bot.trigger_added_events tae
+	LEFT JOIN automation_bot.command_alias alias
+	ON tae.command_address = alias.command_address 
+	LEFT JOIN vulcan2x."transaction" tx
+	ON tx.id = tae.tx_id 
+	LEFT JOIN vulcan2x.block b 
+	ON b.id = tae.block_id 
+	UNION
+	SELECT  tre.id , tre.trigger_id , tre.cdp_id, tre.log_index , tx.hash , b."number", b."timestamp" ,'removes' AS event_type, tae.command_address, alias.kind  from automation_bot.trigger_removed_events tre
```
Comment (adamskrodzki) [inline src/borrow/migrations/022-automation-events-ddl-kind-type.sql:18]: not so important, but if 'added' and 'executed'  then i guess 'removed' not 'removes'

File: src/config.goerli.ts [hunk 7]
```diff
@@ -220,5 +224,16 @@ export const config: UserProvidedSpockConfig = {
     },
   },
   addresses,
-  onStart: () => {},
+  onStart: async (services) => {
+   await initializeCommandAliases(services);
+  },
 };
+
+async function initializeCommandAliases(services: any) {
```
Comment (adamskrodzki) [inline src/config.goerli.ts:232]: this function should not be here, in some common-utils instead

File: src/config.goerli.ts [hunk 8]
```diff
@@ -220,5 +224,16 @@ export const config: UserProvidedSpockConfig = {
...
+  const promises = Object.entries(commandMapping).map(([key, value]) => {
+    const query = `insert  into automation_bot.command_alias(command_address , kind) values ('${key}', '${value}');`;
+    const result = services.db.none(query);
+    return result;
+  });
+  await Promise.all(promises);
```
Comment (adamskrodzki) [inline src/config.goerli.ts:238]: not sure if it will behave correctly for multiple entries if the same object db is reused in parallel.

Comment (adamskrodzki) [reply src/config.goerli.ts:238]: I would use sequential for loop instead, maybe there is some concise syntax with forEach but I'm not sure.

File: src/config.goerli.ts [hunk 9]
```diff
@@ -220,5 +224,16 @@ export const config: UserProvidedSpockConfig = {
...
   },
   addresses,
-  onStart: () => {},
+  onStart: async (services) => {
+   await initializeCommandAliases(services);
+  },
 };
+
+async function initializeCommandAliases(services: any) {
+  const promises = Object.entries(commandMapping).map(([key, value]) => {
```
Comment (adamskrodzki) [inline src/config.goerli.ts:233]: take commandMapping as a parameter do not rely on scope.

**Review by adamskrodzki — CHANGES_REQUESTED** [review]

File: src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql (hunk 3 above)
Comment (adamskrodzki) [reply src/borrow/migrations/022-automation-events-ddl-dml-kind-type.sql]: (triggerId,event_type) is unique key here but do we actually need uniq key in a view?

File: src/config.goerli.ts (hunk 7 above)
Comment (paszkowskiDamian) [inline src/config.goerli.ts:232]: The type of services is probably `Services`

File: src/config.goerli.ts (hunk 8 above)
Comment (paszkowskiDamian) [reply src/config.goerli.ts:238]: You can insert multiply rows like that
``` 
// vatTransformer.ts
const cs = new services.pg.helpers.ColumnSet(
        [
          'kind',
          'collateral_amount',
          'dai_amount',
          'urn',
          'timestamp',
          'tx_id',
          'block_id',
          'log_index',
          'rate',
          'ilk',
        ],
        {
          table: {
            table: 'events',
            schema: 'vault',
          },
        },
      );

      const query = services.pg.helpers.insert(events, cs);
      await services.tx.none(query);
```

Comment (paszkowskiDamian) [inline src/config.goerli.ts:238]: Remember to either remove old entries or handle conflicts, postgres has this feature
```
INSERT INTO my_table (column1, column2)
VALUES
    (value1, value2),
    (value3, value4),
    (value5, value6),
    (value7, value8)
ON CONFLICT DO NOTHING;
```

**Pull request merged**